internal/pkg/utils/deepcopy/step.go

```diff
@@ -0,0 +1,74 @@
+package deepcopy
+
+import (
+	"fmt"
+	"strings"
+)
+
+type Steps []fmt.Stringer
+
```

Comment:
`deepcopy` klonuje datove struktury do hlbky, a vie ich pri tom aj modifikovat.
Pouzivame to na viacero ucelov.

`Steps`, tj. kde sa v datovych strukturach aktualne nachadzame - boli reprezentovane ako string.

To mi uz nestaci, a tak som vytvoril samostatne struktury `TypeStep, PoinsterStep, MapKeyStep ...`.

Prevod do `String` vrati rovnake hodnoty, nebolo potrebne menit testy:
https://github.com/keboola/keboola-as-code/blob/5ba4eb1414dd57dd2e56779c2ff78cab15732f08/internal/pkg/utils/deepcopy/deepcopy_test.go#L94-L142

Reply:
Pouzijem to na nahradzovanie `inputs` pri vytvarani sablony, ... tam nestaci nahradenie podla hodnoty, ... ale je potrebne vediet presnejsie, kde sa nachadzam, ... pod akym klucom je hodnota ulozena.